File: tut/src/service/adder_client.py
#!/usr/bin/env python3

import rospy as ros
from tut.srv import *


def main():
    ros.init_node("adder_clientpy")
    ros.sleep(1)

    client = ros.ServiceProxy('/adder', Adder)
    ros.wait_for_service('/adder')

    while (not ros.is_shutdown()):
        request = AdderRequest()
    
        ros.loginfo("Please input two numbers: ")
        try:
            request.a.num, request.b.num = map(int, input().split())
        except (ValueError):
            continue;
        
        response = client(request)
        ros.loginfo(f"Sum = {response.sum.num}\n")


if (__name__ == "__main__"):
    main()

File: tut/src/service/adder_client.cpp
#include <ros/ros.h>
#include "tut/Adder.h"
#include <iostream>


int main(int argc, char **argv)
{
    ros::init(argc, argv, "adder_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<tut::Adder>("/adder");
    tut::Adder adder_msg;

    while (ros::ok())
    {
        client.waitForExistence();
        ros::Duration(1).sleep();
        ROS_INFO("Please input two numbers:");
        std::cin >> adder_msg.request.a.num >> adder_msg.request.b.num;
        client.call(adder_msg);
        
        ROS_INFO("Sum = %d\n", adder_msg.response.sum.num);
    }

    return 0;
}
